src/intrinsics/StringPrototype.mjs

```diff
@@ -154,29 +159,49 @@ function StringProto_includes([searchString = Value.undefined, position = Value.
 
 // 21.1.3.8 #sec-string.prototype.indexof
 function StringProto_indexOf([searchString = Value.undefined, position = Value.undefined], { thisValue }) {
+  // 1. Let O be ? RequireObjectCoercible(this value).
   const O = Q(RequireObjectCoercible(thisValue));
+  // 2. Let S be ? ToString(O).
   const S = Q(ToString(O)).stringValue();
-  const searchStr = Q(ToString(searchString)).stringValue();
+  // 3. Let searchStr be ? ToString(searchString).
+  const searchStr = Q(ToString(searchString));
+  // 4. Let pos be ? ToInteger(position).
   const pos = Q(ToInteger(position));
+  // 5. Assert: If position is undefined, then pos is 0.
   Assert(!(position === Value.undefined) || pos.numberValue() === 0);
+  // 6. Let len be the length of S.
   const len = S.length;
+  // 7. Let start be min(max(pos, 0), len).
   const start = Math.min(Math.max(pos.numberValue(), 0), len);
-  const searchLen = searchStr.length;
-  let k = start;
-  while (k + searchLen <= len) {
-    let match = true;
-    for (let j = 0; j < searchLen; j += 1) {
-      if (searchStr[j] !== S[k + j]) {
-        match = false;
-        break;
+  // 8. Let searchLen be the length of searchStr.
+  const searchLen = searchStr.stringValue().length;
+  // https://tc39.es/proposal-string-replaceall/#sec-string.prototype.indexof
+  if (surroundingAgent.feature('String.prototype.replaceAll')) {
+    // Let position be ! StringIndexOf(S, searchString, start).
+    // TODO: fix bug in spect text.
+    position = X(StringIndexOf(new Value(S), searchStr, start));
```

Comment:
/cc @mathiasbynens I think that's a bug in the spec.

Reply:
Sorry, I lack context here; what's the bug? Would you mind submitting a patch against https://github.com/tc39/proposal-string-replaceall?

Reply:
The relevant line is here: https://github.com/tc39/proposal-string-replaceall/blob/33f5e50dd27ade44a52a586933d8ec32bbee164b/spec.html#L131

Reply:
I opened an issue: https://github.com/tc39/proposal-string-replaceall/issues/36

Reply:
I see. Good catch! Thank you for filing the bug :)